### ISSUE TITLE
az post upgrade apim platform version to stv2 err saying it already is that version

### ISSUE THREAD

**Issue by davildaran** [Azure/azure-cli#26294]
New issue created because `0` results when searching for `migrateToStv2`.

**Related command**
```
$ az rest --method post --uri "/subscriptions/<subId>/resourceGroups/<resourceGroup>/providers/Microsoft.ApiManagement/service/<apim-name>/migrateToStv2?api-version=2022-08-01"
```

**Describe the bug**
Above command returns the following

```
Bad Request({"error":{"code":"NotSupported","message":"Service is already on stv2.","details":null,"innerError":null}})
```

 despite it is still `stv1` for the Platform Version.
<img width="225" alt="Screenshot 2023-04-28 at 12 37 03 PM" src="https://user-images.githubusercontent.com/18368743/235204832-632fae6d-ba6a-42ad-9663-c2383de4e569.png">

```
$ az apim show --name <name> -g <resourceGroup> --query platformVersion --output tsv                               
stv1

```

**To Reproduce**
1. Run the above command on an apim instance in the Developer Tier.

**Expected behavior**
No errors. Migration should happen in 10-15 minutes like it does for our other APIM instancs. The same command is successful at upgrading from `stv1` to `stv2` for (2) other APIM instances we have in the Basic tier 

**Environment summary**
`azure-cli` version `2.47.0 *`
To recap: working on apim instances in the Basic tier. Not working on an APIM instance we have in the Developer tier.

**Additional context**

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by yingru97** [comment]
@davildaran This looks like an issue specific to that service instead of a common issue with the CLI. Please open a support ticket if you more help. Thanks! https://learn.microsoft.com/en-us/azure/azure-portal/supportability/how-to-create-azure-support-request
